tap_google_sheets/client.py

```diff
@@ -116,12 +116,12 @@ def raise_for_error(response):
                 # There is nothing we can do here since Google has neither sent
                 # us a 2xx response nor a response content.
                 return
+            status_code = response.status_code # Get status code of the response
```

Comment:
In the previous, it was trying to get status code from error.code of JSON response. But in many responses, the value of the error key is just a string. So, it was throwing AttributeError: 'str' object has no attribute 'get' because of error.code missing in response. So, here we are retrieving the status code from the response object itself.